glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -257,6 +258,7 @@ open class GleanInternalAPI internal constructor () {
      *
      * @param enabled When true, enable metric collection.
      */
+    @Synchronized
```

Comment:
I wonder if the solution here is to call `initializeCoreMetrics` using `Dispatchers.API.launch` when called outside of the `Glean.initialize`. Have you tried that, @mdboom ?

My concern is that making the whole init `synchronized`, even though off the main thread, might have performance implications. Of course I have no data to back my claim :)

Feel free to ignore this if it feels sketchy, I'm all up for pushing this to release.

Reply:
Yeah, you could be right.  I was going for a solution I was pretty sure to work, rather than the minimal one.  I might spend a little more time trying to write a test to reproduce -- that would give me more confidence that a minimal solution would work.